dashboard/legacy/middleware/files_api.rb

```diff
@@ -428,11 +438,17 @@ def put_file(endpoint, encrypted_channel_id, filename, body)
       # TODO(JillianK): we are temporarily ignoring address share failures because our address detection is very broken.
       # Once we have a better geocoding solution in H1, we should start filtering for addresses again.
       # Additional context: https://codedotorg.atlassian.net/browse/STAR-1361
-      if share_failure && share_failure[:type] != "address"
+      if app_type != 'aichat' && share_failure && share_failure[:type] != "address"
         details_key = share_failure.type == ShareFiltering::FailureType::PROFANITY ? "profaneWords" : "pIIWords"
         details = {details_key => [share_failure.content]}
         return json_bad_request(details)
       end
+      # For now, just check for profanity in aichat.
+      if app_type == 'aichat' && share_failure && share_failure[:type] == ShareFiltering::FailureType::PROFANITY
```

Comment:
>Any harm in also checking PII? Might be something we need to clarify with curriculum/product, but since I imagine we'd also eventually be checking for PII in aichat_controller would it make sense to do it here as well (and not have to have this separate branch?)

Similar to response to comment above, I think perhaps we should first implement profanity checking, and then follow-up with PII when Sam gets back.

>Alternatively, if we want to check profanity only (that might be useful for other labs too), would be good to check the whole list rather than aichat specifically. 

Sounds good - thanks for the suggestion. Will update!

Reply:
Looking into what happens when both profanity and PII are included in the inspected text. If both are included, only PII will be reported according to https://github.com/code-dot-org/code-dot-org/blob/0be01bc8a5b927bf6b221b882dae97739463fe8c/lib/cdo/share_filtering.rb#L136.

Reply:
So I think I may update this method in `share_filtering` since we're interesting in filtering for profanity only at this point.

Reply:
Update: added `find_profanity_failure` in `share_filtering.rb` and updated the checking in `files_api.rb` https://github.com/code-dot-org/code-dot-org/pull/59414/commits/3ba5cb44226da04f29aa19f818af37fdacccd267#diff-7f7b9ffe903479398ea459bc7a01594b25b89baaa2c8fdda6aeb85deb7f9aacb